packages/block-editor/src/components/block-tools/block-contextual-toolbar.js

```diff
@@ -197,31 +87,7 @@ function UnforwardedBlockContextualToolbar(
 			variant={ isFixed ? 'unstyled' : undefined }
 			{ ...props }
 		>
-			{ ! isCollapsed && <BlockToolbar hideDragHandle={ isFixed } /> }
-			{ isFixed && isLargeViewport && blockType && (
-				<ToolbarGroup
-					className={
-						isCollapsed
-							? 'block-editor-block-toolbar__group-expand-fixed-toolbar'
-							: 'block-editor-block-toolbar__group-collapse-fixed-toolbar'
-					}
-				>
-					<ToolbarItem
-						as={ ToolbarButton }
-						ref={ toolbarButtonRef }
-						icon={ isCollapsed ? next : previous }
-						onClick={ () => {
-							setIsCollapsed( ( collapsed ) => ! collapsed );
-							toolbarButtonRef.current.focus();
-						} }
-						label={
-							isCollapsed
-								? __( 'Show block tools' )
-								: __( 'Hide block tools' )
-						}
-					/>
-				</ToolbarGroup>
-			) }
+			<BlockToolbar hideDragHandle={ isFixed } />
```

Comment:
The component here is named `BlockContextualToolbar`. It was never meant to render the "top toolbar". The original idea has been:

 - If `hasFixedToolbar` is false (default), the `BlockContextualToolbar` is rendered, otherwise it's not and you are (consumer of block-editor package, in our case edit-* packages) responsible of rendering the block toolbar by using the `<BlockToolbar />` component directly. 

I know this has changed over time but this PR might be the opportunity to restore that simple behavior.